include/mbedtls/check_config.h

```diff
@@ -66,6 +66,14 @@
 #error "MBEDTLS_HAVE_TIME_DATE without MBEDTLS_HAVE_TIME does not make sense"
 #endif
 
+#if defined(MBEDTLS_ZLIB_SUPPORT) && defined(MBEDTLS_DEPRECATED_WARNING)
```

Comment:
Why do this in `check_config.h` rather than in the header where it's used as usual?

If a user has their own `config.h` that doesn't include `check_config.h`, we'd still want them to see the deprecation warning. (Except if they haven't enabled any feature that depends on this, then we don't care if they learn about the deprecation.) So this should be in `ssl.h`, not in `check_config.h`.

Reply:
I agree with Gilles that `check_config.h` does not look like the best place, for the reasons stated, and also as Simon hints at below, if this file becomes automatically generated, it will be a burden to maintain manual parts in it.

Also, I think the warning should only be issued if `MBEDTLS_SSL_TLS_C` is defined.